### PR TITLE
Joining a game and dealing cards

### PR DESCRIPTION
**Pull request by soulnafein**
Very basic game state machine that simulate
two players joining a game of poker and a
initial hand being dealt to the players

### CODE REVIEW

**Comment by soulnafein** [comment]
@Arkham Is this what you had in mind? Can suggest changes to make it more idiomatic please?

File: lib/pokerwars/game.ex [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    {deck, players} = Enum.reduce(players, {deck, []}, deal_hand)
+    {deck, players} = Enum.reduce(players, {deck, []}, deal_hand)
+
+    %{game | players: players, current_deck: deck}
+  end
+
+  defp deal_card_to_player(deck, player) do
+    {card, deck} = Deck.deal(deck)
+    hand = player.hand ++ [card]
```
Comment (Arkham) [inline lib/pokerwars/game.ex:70]: Probably we should have a player module that just exposes a function called `give_card`, so we don't need to expose the underlying implementation.

File: lib/pokerwars/game.ex [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  defp phase(:waiting_for_players, game, action) do
+    waiting_for_players(action, game)
+  end
+
+  defp phase(:ready_to_start, game, action) do
+    ready_to_start(action, game)
+  end
+
+  defp waiting_for_players({:join, player}, game) do
+    players = game.players ++ [player]
```
Comment (Arkham) [inline lib/pokerwars/game.ex:26]: If we don't care about the order of the players we can do `players = [player | game.players]`

File: lib/pokerwars/game.ex [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  end
+
+  defp deal_hands(game) do
+    game
+    |> clear_table
+    |> deal_two_cards_to_each_player
+  end
+
+  defp clear_table(game) do
+    players = Enum.map(game.players, fn player -> put_in(player.hand, []) end)
```
Comment (Arkham) [inline lib/pokerwars/game.ex:49]: Here it could be nice to have a Player module that exposes a function that clears the hand of the player.

**Review by Arkham — APPROVED** [review]
Code looks great, just left some minor comments, merge away! 👍

**Pull request merged**